components/tools/ToolsDashboard.tsx

```diff
@@ -149,18 +136,30 @@ export default function ToolsDashboard() {
       }
     });
 
-    setToolsList(tempToolsList);
-  };
+    return tempToolsList;
+  }, [isPaid, isAsyncAPIOwner, languages, technologies, categories, searchName]);
+
+  // useEffect to scroll to the opened category when url has category as element id
+  useEffect(() => {
+    const { hash } = window.location;
 
+    if (hash) {
+      const elementID = decodeURIComponent(hash.slice(1));
+      const element = document.getElementById(elementID);
+
+      if (element) {
+        document.documentElement.style.scrollPaddingTop = '6rem';
+        element.scrollIntoView({ behavior: 'smooth' });
+        document.documentElement.style.scrollPaddingTop = '0';
+      }
+    }
+  }, []);
+  // Function to update the list of tools according to the current filters applied
   const clearFilters = () => {
     setOpenFilter(false);
     router.push('/tools', undefined, { shallow: true });
   };
 
-  useEffect(() => {
-    updateToolsList();
-  }, [isPaid, isAsyncAPIOwner, languages, technologies, categories, searchName]);
-
```

Comment:
Why you removed this useEffect? It's important to apply filters on Tools dashboard.

Reply:
I have moved the filter logic into a useMemo hook (please check [here](https://github.com/asyncapi/website/pull/3116/files#diff-064ff49eb74c78a21589d109c2899f3355e8f15a5c1971410776f6065f885b3aL74-L153)). The purpose of using useMemo is [to avoid storing the tools list in React state](https://github.com/asyncapi/website/pull/3116/files#diff-064ff49eb74c78a21589d109c2899f3355e8f15a5c1971410776f6065f885b3aL33). Initially, when we used React state, the tools list took some time to appear after the page loaded, which made scrolling difficult.